test/functional/lib/tests.cpp

```diff
@@ -2342,23 +2347,25 @@ CreateTcpSocket(
     // within the timeout, the listening socket gets closed, which will cancel
     // the accept request.
     //
-    TCP_ACCEPT_THREAD_CONTEXT Ctx;
+    struct TCP_ACCEPT_THREAD_CONTEXT {
+        FNSOCK_HANDLE ListeningSocket;
+        FNSOCK_HANDLE AcceptedSocket;
+    } Ctx;
     Ctx.ListeningSocket = Socket.get();
-    CXPLAT_THREAD_CONFIG ThreadConfig {
-        0, 0, NULL, TcpAcceptFn, &Ctx
-    };
-    CXPLAT_THREAD AsyncThread;
-    TEST_TRUE(CXPLAT_SUCCEEDED(CxPlatThreadCreate(&ThreadConfig, &AsyncThread)));
-    auto DeleteThread = wil::scope_exit([&]
-    {
-        CxPlatThreadDelete(&AsyncThread);
-    });
+
+    CxPlatAsync Async([](void *Context) -> void* {
+        struct TCP_ACCEPT_THREAD_CONTEXT *Ctx = (struct TCP_ACCEPT_THREAD_CONTEXT *)Context;
+        SOCKADDR_INET Address;
+        INT AddressLength = sizeof(Address);
+        Ctx->AcceptedSocket = FnSockAccept(Ctx->ListeningSocket, (SOCKADDR *)&Address, &AddressLength);
+        return nullptr;
+    }, &Ctx);
     auto CloseListener = wil::scope_exit([&]
     {
         Socket.reset();
     });
```

Comment:
Thoughts on compacting this some? Either:
```c++
    auto CloseListener = wil::scope_exit([&] {
        Socket.reset();
    });
```
or
```c++
    auto CloseListener = wil::scope_exit([&] { Socket.reset(); });
```

Reply:
this scope exit should be deleted. I will do this after all tests pass.